### PR TITLE
Updating CDK based lambdas to java 8 corretto

### PR DESCRIPTION
**Pull request by jambo1**
## What does this change?
Updates lambdas using CDK to build their infrastructure to use java 8 corretto

### CODE REVIEW

**Review by kelvin-chappell — APPROVED** [review]
👍

File: cdk/package.json [hunk 1]
```diff
@@ -15,20 +15,20 @@
...
   "dependencies": {
-    "@aws-cdk/core": "1.27.0",
     "@aws-cdk/aws-apigateway": "^1.27.0",
     "@aws-cdk/aws-events": "1.27.0",
     "@aws-cdk/aws-events-targets": "1.27.0",
     "@aws-cdk/aws-iam": "^1.27.0",
-    "@aws-cdk/aws-lambda": "^1.27.0",
+    "@aws-cdk/aws-lambda": "^1.121.0",
     "@aws-cdk/aws-s3": "^1.27.0",
+    "@aws-cdk/core": "1.27.0",
```
Comment (kelvin-chappell) [inline cdk/package.json:31]: Shouldn't this be `^1.121.0` as well?

File: cdk/package.json [hunk 2]
```diff
@@ -15,20 +15,20 @@
     "@aws-cdk/assert": "1.27.0",
     "@types/jest": "^24.0.22",
     "@types/node": "10.17.5",
+    "aws-cdk": "^1.121.0",
```
Comment (kelvin-chappell) [inline cdk/package.json:18]: Is this necessary if `aws-cdk/core` is in the dependency list?

File: cdk/package.json (hunk 1 above)
Comment (jambo1) [reply cdk/package.json:31]: I've updated both to 1.27 and I'll keep this PR on draft until I get it tested on code, I'm still getting used to github apologies for any notifications!

File: handlers/sf-move-subscriptions-api/cfn.yaml [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+        Statement:
+          - Action: execute-api:Invoke
+            Effect: Allow
+            Principal: "*"
+            Resource: execute-api:/*/*/*
+          - Action: execute-api:Invoke
+            Condition:
+              NotIpAddress:
+                aws:SourceIp:
+                  # is this okay or do we need the below?
```
Comment (kelvin-chappell) [inline handlers/sf-move-subscriptions-api/cfn.yaml:117]: I think you can get rid of the commented-out section but I'm not sure.  Would need to trial it in Code.

File: handlers/digital-voucher-cancellation-processor/cfn.yaml [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+            - ""
+            - - membership/
+              - Ref: Stage
+              - /digital-voucher-cancellation-processor/digital-voucher-cancellation-processor.jar
+      Handler: com.gu.digital_voucher_cancellation_processor.Handler::handle
+      Role:
+        Fn::GetAtt:
+          - DigitalVoucherCancellationProcessorFnRole9BC677A8
+          - Arn
+      Runtime: java8
```
Comment (kelvin-chappell) [inline handlers/digital-voucher-cancellation-processor/cfn.yaml:177]: Will need updating

**Review by kelvin-chappell — APPROVED** [review]
If it all deploys OK in Code 👍
 
I imagine the Teamcity scripts might need updating as well.

**Pull request merged**